finality-aleph/src/finalization.rs

```diff
@@ -77,194 +80,133 @@ where
     }
 }
 
-/// Given hash `last_finalized` and `AlephDataFor` `new_data` of two blocks, returns
-/// Some(new_data) if the block hash represented by new_data is a descendant of last_finalized
-/// (and the new_data.number is correct). Otherwise it outputs None.
-pub(crate) fn should_finalize<B, C>(
-    last_finalized: B::Hash,
-    new_data: AlephDataFor<B>,
+/// Checks if the data is available and that the block numbers match with the ones provided.
```

Comment:
again, the 'correctness' is not full - we do not check here whether this branch is based on some finalized block or goes in parallel to a finalized block; this shouldn't be so expensive to add

Reply:
I changed the comment and generally improved the code in this file quite a bit. My intention is for this method to only check block availability, not whether the proposal will actually result in finalizing its top block.